app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb

```diff
@@ -12,8 +12,9 @@ def create
     permissions.authorise(identifier)
     Auditor::Audit.new(nil, 'dfe-sign-in.authentication.success', current_session_id).log_without_association
 
-    if permissions.school_urn.present?
-      update_session(permissions.school_urn)
+    if permissions.all_permissions.any?
```

Comment:
What would happen if the `selected_school_urn` was present but not included within `permissions.all_permissions`?

Reply:
As long as they had some permission in TVA it would sign them in using the selected school.

I think this makes sense as we're authorising users not schools? 

If we were to check the school urn was present in TVA we could get to a situation where a user has 10 schools in DfE Sign In and only 5 in TVA... how do we keep those in sync?

Reply:
The current design of TVA provides us granular control in granting and revoking access between users and schools, we _are_ authorising users but the only thing we care about is their access to an individual school. 

Examples where losing this would cause confusion:
- Individual hiring staff should only have access to a subset of schools they are responsible for
- Hiring staff moves between organisations but continue to have full access to listings they should no longer be managing
- Hiring staff is added to another organisation in error (by another service perhaps) and has full access to listings they should not be managing
- Hiring staff are attached to multiple organisations, valid schools and other inappropriate organisations eg. Universities.

In general I think this approach would introduce a grey area of uncertainty and we'd be less confident in knowing what users had access to what school at any point in time.

You're right these is a synchronisation problem to solve. I was thinking that until DfE Sign-in can replace TVA we could get by with the quick and dirty solution of noting down the log output when an association could not be made during our invitations and manually creating it through the support console. If that cannot be achieved, we talk to DSI and GIAS to find out why that school urn is not yet in DSI. It may be that we have to wait a day or so before that association can be made.